dhall/tests/Dhall/Test/TypeInference.hs

```diff
@@ -41,24 +48,60 @@ successTest prefix = do
                  -- fails due to `Expr`'s 'Eq' instance, which inherits the
                  -- @NaN /= NaN@ inequality from 'Double'.
                  typeInferenceDirectory </> "success/unit/AssertNaN"
+
+               , typeInferenceDirectory </> "success/prelude" -- TODO
+
+                 -- https://github.com/dhall-lang/dhall-haskell/issues/1274
+               , typeInferenceDirectory </> "success/unit/RecordProjectionByTypeJudgmentalEquality"
                ]
 
     Test.Util.testCase prefix skip $ do
-        value <- expr "A.dhall"
+        let actualCode = Test.Util.toDhallPath (prefix <> "A.dhall")
```

Comment:
As demonstrated in #1274, we mustn't access the source expression via an import since that can change the type. 
Instead, we'll have to simply read the file and then resolve it relative to its directory.